src/bun.js/node/node_fs.zig

```diff
@@ -2633,14 +2633,30 @@ pub const NodeFS = struct {
                     if (size == 0) {
                         // copy until EOF
                         while (true) {
-
                             // Linux Kernel 5.3 or later
-                            const written = linux.copy_file_range(src_fd, &off_in_copy, dest_fd, &off_out_copy, std.mem.page_size, 0);
+                            var written = linux.copy_file_range(src_fd, &off_in_copy, dest_fd, &off_out_copy, std.mem.page_size, 0);
+
                             if (ret.errnoSysP(written, .copy_file_range, dest)) |err| {
-                                // TODO: handle EXDEV
-                                // seems like zfs does not support copy_file_range across devices
-                                // see https://discord.com/channels/876711213126520882/876711213126520885/1006465112707698770
-                                return err;
+                                switch (err.getErrno()) {
+                                    .XDEV => {
+                                        var buffer: [4096]u8 = undefined;
+
+                                        var bytes_read = switch (Syscall.read(src_fd, &buffer)) {
+                                            .result => |result| result,
+                                            .err => |_err| return Maybe(Return.CopyFile){ .err = _err.withPath(src) },
+                                        };
+
+                                        if (bytes_read == 0) break;
+
+                                        var slice = buffer[0..bytes_read];
+
+                                        written = switch (Syscall.write(dest_fd, slice)) {
+                                            .result => |result| result,
+                                            .err => |_err| return Maybe(Return.CopyFile){ .err = _err.withPath(src) },
+                                        };
+                                    },
+                                    else => return err,
+                                }
```

Comment:
Can we reduce this duplicate code?

https://github.com/TiranexDev/bun/blob/92474f3302454fd3a280a7b3c0a1695ffc33a251/src/bun.js/node/node_fs.zig#L2633-L2698

Reply:
cc @Jarred-Sumner

Reply:
I'll see what I can do meanwhile